// user.proto
syntax = "proto3";

package user;
option go_package = "./user";

service UserService {
  rpc GetUser(UserRequest) returns (UserResponse) {}
  rpc ListUsers(Empty) returns (UserList) {}
}

message Empty {}

message UserRequest {
  int32 user_id = 1;
}

message UserResponse {
  int32 user_id = 1;
  string name = 2;
  string email = 3;
}

message UserList {
  repeated UserResponse users = 1;
}